lib/src/utils.dart

```diff
@@ -104,3 +105,84 @@ String getResponseError(Response res) {
 
 Future<String> ownerToAddress(String owner) async => encodeBytesToBase64(
     await sha256.hash(decodeBase64ToBytes(owner)).then((res) => res.bytes));
+
+Uint8List longTo32ByteArray(int long) {
+  // we want to represent the input as a 8-bytes array
+  final byteArray = [
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+    0,
+  ];
+
+  for (var index = 0; index < byteArray.length; index++) {
+    final byte = long & 0xff;
+    byteArray[index] = byte;
+    long = (long - byte) ~/ 256;
+  }
+
+  return Uint8List.fromList(byteArray);
+}
+
+Uint8List shortTo2ByteArray(int long) {
+  if (long > (2 ^ (32 - 1))) throw Exception('Short too long');
+
+  var byteArray = [0, 0];
+
+  for (var index = 0; index < byteArray.length; index++) {
+    final byte = long & 0xff;
+    byteArray[index] = byte;
+    long = (long - byte) ~/ 256;
+  }
+
+  return Uint8List.fromList(byteArray);
+}
+
+int byteArrayToLong(Uint8List byteArray) {
+  var value = 0;
+  for (var i = byteArray.length - 1; i >= 0; i--) {
+    value = value * 256 + byteArray[i];
+  }
+  return value;
+}
+
+Uint8List longTo8ByteArray(int long) {
+  // we want to represent the input as a 8-bytes array
+  final byteArray = [0, 0, 0, 0, 0, 0, 0, 0];
```

Comment:
```suggestion
  final byteArray = new List(8);
```

Reply:
See answer above. I don't know if I should change this. I think seeing the list this way sort of gives you an immediate idea what's happening, but that could just be me.

Reply:
It's not so hard for the dev to count 2 or 8 zeroes, but when it raises to 32 that's less trivial. See my comment above using `List.fill` as alternative to `List.generate`